pkg/virtual/syncer/transformations/helpers.go

```diff
@@ -0,0 +1,107 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package transformations
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	"github.com/kcp-dev/kcp/pkg/apis/workload/v1alpha1"
+)
+
+// GetSyncing gathers, for each SyncTarget, all the information related
+// to the Syncing of the resource to this SyncTarget.
+// This information comes from labels and annotations.
+// Keys in the returns map are SyncTarget keys.
+func GetSyncing(upstreamResource metav1.Object) (map[string]SyncTargetSyncing, error) {
+	labels := upstreamResource.GetLabels()
+	syncing := make(map[string]SyncTargetSyncing, len(labels))
+	annotations := upstreamResource.GetAnnotations()
+	for labelName, labelValue := range labels {
+		if strings.HasPrefix(labelName, v1alpha1.ClusterResourceStateLabelPrefix) {
+			syncTarget := strings.TrimPrefix(labelName, v1alpha1.ClusterResourceStateLabelPrefix)
+			syncTargetSyncing := SyncTargetSyncing{
+				ResourceState: v1alpha1.ResourceState(labelValue),
+			}
+			if deletionAnnotation, exists := annotations[v1alpha1.InternalClusterDeletionTimestampAnnotationPrefix+syncTarget]; exists {
+				var deletionTimestamp metav1.Time
+				if err := deletionTimestamp.UnmarshalText([]byte(deletionAnnotation)); err != nil {
+					return nil, fmt.Errorf("parsing of the deletion annotation for sync target %q failed: %w", syncTarget, err)
+				} else {
+					syncTargetSyncing.DeletionTimestamp = &deletionTimestamp
+				}
+			}
+			if finalizersAnnotation, exists := annotations[v1alpha1.ClusterFinalizerAnnotationPrefix+syncTarget]; exists {
+				syncTargetSyncing.Finalizers = finalizersAnnotation
+			}
+			syncing[syncTarget] = syncTargetSyncing
+		}
+	}
+	return syncing, nil
+}
+
+// getOverriddenSyncerViewFields builds a map whose keys are summarizing field keys,
+// and values are the overridden values of the corresponding fields for this SyncTarget.
```

Comment:
the overridden or overriding values?

Reply:
Now that you say it, it might be `overriding`

Reply:
I updated the code, removing this word when not useful, and changng it when the wrong one was used.